packages/protocol/contracts-0.8/governance/Validators.sol

```diff
@@ -1508,6 +1509,19 @@ contract Validators is
     return true;
   }
 
+  function _sendValidatorPaymentIfNecessary(address validator) private {
+    if (isL2()) {
+      getEpochManager().sendValidatorPayment(validator);
+    }
+  }
+
+  function _sendValidatorGroupPaymentsIfNecessary(ValidatorGroup storage group) private {
```

Comment:
should there be an isL2 check in this function as well? this would avoid going thru the loop. 

Alternatively, it could be implemented when the function is used.